docs/reference/module-api/module-overview.md

```diff
@@ -838,19 +838,82 @@ module.exports = {
 };
 ```
 
-### `queries(self)`
+### `queries(self, query)`
+
+The `queries` function is part of the Apostrophe API for querying the database. It takes two arguments: the module (`self`) and the query that will be called (`query`). It must return an object. That object can have two properties:
+<!-- TODO: Link to a real guide on using queries when available. This is simply reference. -->
+
+| `queries` properties | Description |
+| ---- | ---- |
+| [`builders`](#builders) | An object of "builders," or chainable methods, that refine the query or otherwise change its behavior prior to execution. |
+| [`methods`](#methods) | An object of method functions that execute some action on the query. |
+
+#### `builders`
+
+Query builders are defined as objects with a set of properties available to them. Builders often take an argument or use a default value.
+<!-- TODO: Link to a reference of core builders when available. -->
+
+| Builder properties | Description |
+| ---- | ---- |
+| `def` | The default value for the builder. |
+| `launder` | A function used to validate values passed as arguments when `applyBuildersSafely` is called on the query. Returns `true` if valid.<br />This is required to use the builder in a REST API query string. |
+| `choices` | A function returning an array of "choice" objects with `value` and `label` properties. This is returned from the `toChoices` query method if set. |
+| `set` | A function called when the builder is invoked instead of the default `query.set` method. The `set` function should include running the `query.set` after other work. |
+| `prefinalize` | A function to run before any builder `finalize` steps. Used to alter the query with other builders. |
+| `finalize` | A function to run at the end of the query building phase, prior to being processed by the database. Used to alter the query with other builders. |
+| `after` | A function run to mutate an array of queried items passed in as an argument. The `after` function should include checking for validity prior to mutating. |
```

Comment:
I don't understand where this came from: "The `after` function should include checking for validity prior to mutating."

`after` takes stuff that came back from mongo and does extra work like loading widgets or relationships.

Reply:
I mean that you should check that the builder was used.

```
        relationships: {
          def: true,
          async after(results) {
            if (!query.get('relationships')) {
              return;
            }`
```

Reply:
OK. "validity" implies something more I think. How about "the after function should use `query.get` to confirm that the work should be done."